src/mono/mono/mini/mini-arm64.c

```diff
@@ -3313,7 +3371,7 @@ get_type_size_macro (MonoTypeEnum type)
 	case MONO_TYPE_R8:
 		return TYPE_F64;
 	default:
-		g_assert_not_reached ();
+		return TYPE_I8;
```

Comment:
Which type would hit default case?

Reply:
On the callsite, you have `get_type_size_macro (ins->inst_c1)`. I don't think the field `ins->inst_c1` was assigned to a meaningful value. That's why when calling `get_type_size_macro`, you encountered `MONO_TYPE_END`, which is basically `0`. IMO, the right thing to do here is to leave default case as `g_assert_not_reached ();`. On the callsite, do something like this:
```
const int _t;
const gboolean _f;
if (ins->opcode != OP_XZERO) {
        _t = get_type_size_macro (ins->inst_c1);
    	_f = is_type_float_macro (ins->inst_c1);
} else {
        _t = TYPE_I8;
        _f = FALSE;
}
```

Reply:
Since OP_XZERO is no longer handled by the table, the assert has been restored.